### DISCUSSION TITLE
Rubygems package registry checksum issue?

### DISCUSSION THREAD

**Discussion by sheck** [community/community#86633] [Packages] [answered]
### Select Topic Area

Bug

### Body

I'm seeing issues with github package registry for private rubygems that look similar to #47659

As of 9am pacific today it appeared to be working, and around 11am pacific I noticed the same issue mentioned in the linked discussion. We're seeing it for multiple packages, both new and existing releases.

```
Bundler::HTTPError The checksum of /info/<GEM NAME> does not match the checksum provided by the server! Something is wrong (local checksum is "\"<SOME SHA>\"", was expecting "\"<A DIFFERENT SHA>\"")
```

**Comment by AMathews** [comment]
Can't provide any answers but you are not alone. We are experiencing it also. First noticed the problem over an hour ago when CI builds started failing for checksum mismatches.

**Comment by dnorth98** [comment]
Adding our voice to this issue which is breaking all our CI/CD builds.  We have seen this before last year - maybe it's an annual occurrence now?

Last year's issue: 
https://github.com/orgs/community/discussions/47659

**Comment by peterkarman1** [comment]
![image](https://github.com/community/community/assets/9062217/d586c6b7-d9b7-4c4a-b9d4-0aee596c1694)

**Reply by AMathews** [reply]
Yeah, I guess I'll go outside or something...

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExcG5ncDltZjEyZWxvczV6ZjRrbzVndG56cXB2NWJ5NTZoMjVhenZ4dCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/9uIwA1FTfNTC8p4c4l/giphy.gif)

**Comment by euge** [comment]
Having same issue here, also affecting CI/CD

**Comment by agirlnamedsophia** [comment]
Also seeing this issue, also affecting CI builds.

**Reply by njyrbecker** [reply]
Same here.  CI builds and our dev environments all experiencing same issue for all packages hosted via github private repos.  All were working fine yesterday.

**Comment by ethangarofolo-wsgr** [comment]
Same. Unable to pull gems.

**Comment by milesstanfield** [comment]
same. dead in the water

**Comment by Cookuricoo1** [comment]
Same

**Comment by kellytowle** [comment]
Same

**Comment by scub** [comment]
Hello frenz

![image](https://github.com/community/community/assets/991850/c6385497-89ff-4841-a949-d058ab29d2c3)

**Reply by tgnla** [reply]
:'(

**Reply by Gbarnett374** [reply]
😢

**Comment by iamwhil** [comment]
Creating more noise for this. 🔔

**Reply by emccullough-carerev** [reply]
Yo, hey Whil!

**Reply by JamesMasonRC** [reply]
:woo: !

**Reply by RomanHargraveRC** [reply]
🐦

**Reply by rc-iwoodley** [reply]
Evan!

**Reply by iamwhil** [reply]
Evan! 
It seems you landed here too! 
What a fun day.

**Reply by emccullough-carerev** [reply]
Lol hey everyone! Glad we can get together and enjoy our inaccessible ruby gems!

**Reply by Jarvl** [reply]
Hey there peeps long time no see

**Reply by JamesMasonRC** [reply]
It's a reunion!

**Comment by euge** [comment]
For anyone here stuck, it seems `bundle install --full-index` is able to work around this issue. Definitely not the solution, but a temporary fix.

**Reply by BytewaveMLP** [reply]
Doesn't help in automated environments like Heroku, where you can't control the bundler flags in use. :(

**Reply by neebs12** [reply]
yes! this fixed it for us 😭

**Reply by smudge** [reply]
Worth noting that, as far as I can tell, [this forces bundler to use the OLD gems APIs](https://github.com/rubygems/rubygems/issues/3806#issuecomment-661094651), which may not be desirable if the checksums are truly in a mismatched state (I'm not sure if the old APIs provide the same degree of cross-check/validation on gem install).

**Reply by minionOfZuul** [reply]
AFAICT this is more a problem with http caching digests (etags, etc) than with gem archive checksums.

**Comment by hoenth** [comment]
One more voice in the chorus. CI/CD builds busted, as well as Github Actions.

**Reply by hoenth** [reply]
We too were able to get CI/CD builds going again with `bundle install --full-index`. But not as easy to do the same with GH Actions

**Reply by kourylape** [reply]
Same. It fixes the build step, but still blocks other actions.

**Comment by kourylape** [comment]
+1 

We too are seeing this.

**Reply by emccullough-carerev** [reply]
Hey Koury! Seeing all sorts of former coworkers here! 😂

**Comment by tinaheidinger** [comment, marked as answer]
Hello from GitHub Packages - we're looking into this!

Edit: We just deployed a fix, installing RubyGems packages should now work as expected

**Comment by kpaurora** [comment]
It seems to work again - at least on my end.

**Reply by gauravs** [reply]
~~Same here!~~
Edit: Still broken :(

**Reply by areeseburnett** [reply]
We're still seeing the error.

**Reply by smudge** [reply]
Still seeing the error here as well.

**Reply by JWShuff** [reply]
Same here, wrangling through it, experimenting with the setup ruby action to see if we can work around it in the meanwhile.

**Reply by jeremiahrose** [reply]
Still happening here

**Reply by doanphuctam** [reply]
Still seeing checksum error.

**Comment by smudge** [comment]
I was _just_ able to bundle install again.

Edit: Yup, seems to be working again across a few projects.

**Reply by emccullough-carerev** [reply]
Also seeing bundle installs work again in GitHub Actions at my org

**Reply by milesstanfield** [reply]
same. bundle installs working again at my org for dozens of ci/cd pipelines that were broken before
